cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,60 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfiler::serialize(std::ostream & outstream) const
```

Comment:
It's better to keep the serialization code outside the object to be serialized.  Perhaps try to make a `CallProfilerSerializer`.  What do you think?

Reply:
It looks good to me.
It can make ```CallProfiler``` compact.

Reply:
@ThreeMonth03 I think it's better to make the json parser as a separate class, so later other classes can utilize the json parser as well. Let me know if you would like to implement the generic json parser. I can help implement the parser if you don't have time.

Reply:
> @ThreeMonth03 I think it's better to make the json parser as a separate class, so later other classes can utilize the json parser as well. Let me know if you would like to implement the generic json parser. I can help implement the parser if you don't have time.

I think I'm unable to implement the generic json parser. 
@tigercosmos  Could you give me a hand?